clinica/pipelines/statistics_volume/statistics_volume_utils.py

```diff
@@ -45,26 +53,171 @@ def get_group_1_and_2(tsv, contrast):
+    """Creates the spm output folder. If it already exists, it is deleted and created again.
+
+    Parameters
+    ----------
+    current_model: str
+        Path to the matlab files with all the @TEXT replaced with the correct names
+
+    Returns
+    -------
+    output_folder: str
+        Path to the spm output_folder
     """
-        Create the matlab .m file for the instantiation of the 2-sample t-test model in SPM
-
-    Args:
-        tsv: (str) path to the tsv file containing information on subjects/sessions with all covariates
-        contrast: (str) name of a column of the tsv
-        idx_group1: (list of int) list of indexes of first group
-        idx_group2: (list of int) list of indexes of second group
-        file_list: List of files used in the statistical test. Their order is the same as it appears on the tsv file
-        template_file: (str) path to the template file used to generate the .m file
-
-    Returns:
-        current_model: (str) path to the matlab files with all the @TEXT replaced with the correct names
-        covariates: list of str with the names of covariates
+    from os import mkdir
+    from os.path import abspath, dirname, isdir, join
+    from shutil import rmtree
+
+    output_folder = abspath(join(dirname(current_model), "..", "2_sample_t_test"))
+    if isdir(output_folder):
+        rmtree(output_folder)
+    mkdir(output_folder)
+    return output_folder
+
+
+def set_output_and_groups(
+    output_folder: str,
+    current_model: str,
+    file_list: list,
+    idx_group1: list,
+    idx_group2: list,
+    filedata: str,
+) -> None:
+    """Sets path to output and groups in the .m script
+
+    Parameters
+    ----------
+    output_folder: str
+        Path to the folder where the pipelines output will be generated
+    current_model: str
+        Path to the matlab files with all the @TEXT replaced with the correct names
+    file_list: List
+        List of files used in the statistical test. Their order is the same as it appears on the tsv file
+    idx_group1: list of int
+        List of indexes of first group
+    idx_group2: list of int
+        List of indexes of second group
+    filedata:
+        .m script template which is to be modified and then run
+    """
+    import clinica.pipelines.statistics_volume.statistics_volume_utils as utls
+
+    filedata = filedata.replace("@OUTPUTDIR", "'" + output_folder + "'")
+    filedata = filedata.replace(
+        "@SCANS1",
+        utls.unravel_list_for_matlab(
+            [f for i, f in enumerate(file_list) if i in idx_group1]
+        ),
+    )
+    filedata = filedata.replace(
+        "@SCANS2",
+        utls.unravel_list_for_matlab(
+            [f for i, f in enumerate(file_list) if i in idx_group2]
+        ),
+    )
+
+    with open(current_model, "w+") as file:
+        file.write(filedata)
+
+
+def transform_data(current_covar_data: list) -> list:
+    """Transforms data
```

Comment:
This isn't super informative. This function is assuming a list of strings as input and performs some encoding to output a list of numeric values. In addition, there is nothing specific to covariates here, so we can be more generic with the naming.

So maybe something like this:

```python
def convert_to_numeric(input_list: List[str]) -> list:
    """Convert the string values from the input list to numeric values.

    If the values can be casted to floats, then outputs a list of floats.
    If the values are categorical, then performs encoding of the categories.
    ....
```